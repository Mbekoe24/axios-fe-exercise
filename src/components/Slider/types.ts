import { SlideProps } from "components/Slide/types";

export interface ApiResponse extends SlideProps{
  id: string
}

export interface SliderProps {
  data: SlideProps[],
}