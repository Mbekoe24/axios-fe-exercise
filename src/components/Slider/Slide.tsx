import React from "react";
import { SlideProps } from "../../../types";

export const Slide = (props: SlideProps) => {
  return (

    ////// testing if data populates
    <div className="slides">
      <div id="slide-1">
        <a href={props.permaLink}>
          <img src={props.primaryImage} alt="" />
        </a>
        <h1>{props.headline}</h1>
        <span>
          {props.publishedDate} - {props.sectionLabel}
        </span>
      </div>
      <div id="slide-2">2</div>
      <div id="slide-3">3</div>
      <div id="slide-4">4</div>
      <div id="slide-5">5</div>
    </div>
  );
};
