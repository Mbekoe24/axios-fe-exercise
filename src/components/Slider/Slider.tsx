import Slide from "../Slide";
import React from "react";
import { SliderProps, ApiResponse } from "./types";

const Slider = ({ data }: SliderProps) => {
  const handleClick = () => {
    window.open("https://www.axios.com/");
  };
  return (
    <>
      <h1 data-testid="axios-h1">More from Axios.com</h1>
      <div className=" news-slider">
        <div className="slides">
          {data && (
            <>
              {data?.map((item: ApiResponse, index) => {
                return (
                  <Slide
                    key={item.id}
                    htmlId={`slide-${index + 1}`}
                    headline={item.headline}
                    displayName={item.displayName}
                    topicName={item.topicName}
                    altText={item.altText}
                    publishedDate={item.publishedDate}
                    primaryImage={item.primaryImage}
                    permaLink={item.permaLink}
                  />
                );
              })}
            </>
          )}
        </div>
        <div className="pagination-container">
          {data.map((d, i) => {
            return (
              <a className="pagination" key={i} href={`#slide-${i + 1}`}>{`${
                i + 1
              }`}</a>
            );
          })}
        </div>
      </div>
      <button
        data-testid="axios-btn"
        className="axios-button"
        onClick={handleClick}
      >
        Visit Axios.com
      </button>
    </>
  );
};

export default Slider;
