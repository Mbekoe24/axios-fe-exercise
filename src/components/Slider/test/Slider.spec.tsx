import renderer from "react-test-renderer";
import * as React from "react";
import Slider from "../Slider";

describe("Slider", () => {
  it("it matches snapshot", () => {
    const container = renderer
      .create(
        <Slider
          data={[
            {
              htmlId: "slide",
              headline: "{item.headline}",
              displayName: "{item.displayName}",
              topicName: "{item.topicName}",
              altText: "{item.altText}",
              publishedDate: "{item.publishedDate}",
              primaryImage: "{item.primaryImage}",
              permaLink: "{item.permaLink}",
            },
          ]}
        />
      )
      .toJSON();
    expect(container).toMatchSnapshot();
  });
});
