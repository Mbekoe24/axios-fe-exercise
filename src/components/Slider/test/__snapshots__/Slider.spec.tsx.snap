// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> Should render correctly 1`] = `
<h1
  data-testid="axios-h1"
>
  More from Axios.com
</h1>
`;
